Content.Shared/Emag/Systems/EmagSystem.cs

```diff
@@ -82,6 +83,13 @@ public bool DoEmagEffect(EntityUid user, EntityUid target)
         var emaggedEvent = new GotEmaggedEvent(user);
         RaiseLocalEvent(target, ref emaggedEvent);
 
+        //Don't add EmaggedComponent to the borg if only the lock, and not the borg itself, has been emagged.
+        if (TryComp<EmagSiliconLawComponent>(target, out var emagSiliconLawComponent))
```

Comment:
do not put this in emag code, set borg system order instead and do le logic

Reply:
made an OnAttemptEmag event that checks if the borg is locked